statedb_test.go

```diff
@@ -115,6 +130,18 @@ func TestSuite(t *testing.T) {
 		StorageRoot: Account.Root.String(),
 		Removed:     false,
 	}))
+	require.NoError(t, insertStateCID(pool, stateModel{
+		BlockNumber: NonCanonicalBlockNumber4,
+		BlockHash:   BlockHash4.String(),
+		LeafKey:     AccountLeafKey.String(),
+		CID:         AccountCID.String(),
+		Diff:        true,
+		Balance:     Account.Balance.Uint64(),
```

Comment:
```suggestion
		Balance:     big.NewInt(123).Uint64(),
```
Otherwise it matches the canonical account